Tasks/VsTest/vstest.ts

```diff
@@ -73,13 +73,55 @@ catch (error) {
     throw error;
 }
 
-function getTestAssemblies(): Set<string> {
+function getBasePathFromFilter(testAssemblyFilter) {
```

Comment:
If my logic goes right, testCaseFilter input "C:\\Test\abc.dll" is going to fail very badly.
Entire find matching logic itself looks bad to me.